src/stripeLogsView.ts

```diff
@@ -1,15 +1,203 @@
+  private stripeClient: StripeClient;
+  private logTreeItems: StripeTreeItem[];
+  private viewState: ViewState;
+  private logsStdoutStream: stream.Writable | null;
+  private logsStderrStream: stream.Writable | null;
+
+  constructor(stripeClient: StripeClient) {
+    super();
+    this.stripeClient = stripeClient;
+    this.logTreeItems = [];
+    this.logsStdoutStream = null;
+    this.logsStderrStream = null;
+    this.viewState = ViewState.Idle;
+  }
+
+  async startLogsStreaming() {
+    this.setViewState(ViewState.Loading);
+    try {
+      await this.setupStreams();
+      this.setViewState(ViewState.Streaming);
+    } catch (e) {
+      window.showErrorMessage(e.message);
+      this.cleanupStreams();
+      this.setViewState(ViewState.Idle);
+    }
+  }
+
+  stopLogsStreaming() {
+    this.cleanupStreams();
+    this.setViewState(ViewState.Idle);
+  }
+
   buildTree(): Promise<StripeTreeItem[]> {
-    const logStreamItem = new StripeTreeItem('Start API logs streaming', 'openLogsStreaming');
-    logStreamItem.setIcon({
-      dark: Resource.ICONS.dark.terminal,
-      light: Resource.ICONS.light.terminal,
-    });
+    const streamingControlItemArgs = (() => {
+      switch (this.viewState) {
+        case ViewState.Idle:
+          return {
+            label: 'Start streaming API logs',
+            command: 'startLogsStreaming',
+            iconId: 'play-circle',
+          };
+        case ViewState.Loading:
+          return {
+            label: 'Starting streaming API logs...',
+            command: 'stopLogsStreaming',
+            iconId: 'loading',
+          };
+        case ViewState.Streaming:
+          return {
+            label: 'Stop streaming API logs',
+            command: 'stopLogsStreaming',
+            iconId: 'stop-circle',
+          };
+      }
+    })();
+
+    const streamingControlItem = this.createItemWithCommand(streamingControlItemArgs);
+
+    const treeItems = [streamingControlItem];
+
+    if (this.logTreeItems.length > 0) {
+      const logsStreamRootItem = new StripeTreeItem('Recent logs');
+      logsStreamRootItem.children = this.logTreeItems;
+      logsStreamRootItem.expand();
+      treeItems.push(logsStreamRootItem);
+    }
+
+    return Promise.resolve(treeItems);
+  }
+
+  private createItemWithCommand({
+    label,
+    command,
+    iconId,
+  }: {
+    label: string;
+    command?: string;
+    iconId?: string;
+  }) {
+    const item = new StripeTreeItem(label, command);
+    if (iconId) {
+      item.iconPath = new ThemeIcon(iconId);
+    }
+    return item;
+  }
+
+  private async setupStreams() {
+    const stripeLogsTailProcess = await this.stripeClient.getOrCreateCLIProcess(
+      CLICommand.LogsTail,
+      ['--format', 'JSON'],
+    );
+    if (!stripeLogsTailProcess) {
+      throw new Error('Failed to start `stripe logs tail` process');
+    }
+
+    stripeLogsTailProcess.on('close', this.cleanupStreams);
+    stripeLogsTailProcess.on('error', this.cleanupStreams);
```

Comment:
fwiw, you should probably only need to listen on `close` since you don't need to handle any special case for errors